cmd/csi-addons/replication.go

```diff
@@ -0,0 +1,245 @@
+/*
+Copyright 2023 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/csi-addons/kubernetes-csi-addons/internal/proto"
+	"github.com/csi-addons/kubernetes-csi-addons/internal/sidecar/service"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type VolumeReplicationBase struct {
+	// inherit Connect() and Close() from type grpcClient
+	grpcClient
+
+	parameters      map[string]string
+	secretName      string
+	secretNamespace string
+	volumeID        string
+}
+
+func (rep *VolumeReplicationBase) Init(c *command) error {
+	rep.parameters = make(map[string]string)
+	rep.parameters["clusterID"] = c.clusterid
+	if rep.parameters["clusterID"] == "" {
+		return fmt.Errorf("clusterID not set")
+	}
+
+	secrets := strings.Split(c.secret, "/")
+	if len(secrets) != 2 {
+		return fmt.Errorf("secret should be specified in the format `namespace/name`")
+	}
+	rep.secretNamespace = secrets[0]
+	if rep.secretNamespace == "" {
+		return fmt.Errorf("secret namespace is not set")
+	}
+
+	rep.secretName = secrets[1]
+	if rep.secretName == "" {
+		return fmt.Errorf("secret name is not set")
+	}
```

Comment:
new errors.New() when formatting is not required

Reply:
Hi Madhu, just a slight confusion, did you meant to remove this below error 
>>return fmt.Errorf("secret should be specified in the format `namespace/name`")

Reply:
No, I meant use errors.New() instead of fmt.Errorf

Reply:
Thanks Madhu, Updated.